dist/index.d.ts

```diff
@@ -48,11 +54,16 @@ declare global {
 
   export interface GPUExtent3DDict {
     width: number;
-    height: number;
-    depthOrArrayLayers: number;
+    height?: number;
+    depthOrArrayLayers?: number;
   }
   export type GPUExtent3D = number[] | GPUExtent3DDict;
 
+  interface GPUExtent3DDictStrict extends GPUExtent3DDict {
```

Comment:
ok this is confusing me but I haven't been following the gpuweb PRs so much.
so 1390 + 1152 + 1441 allows you to say either "depth" OR "depthOrArrayLayers" ?

Reply:
no, `depth` doesn't exist anymore. The IDL completely ignores it, which means if you DO provide a `depth` you'll end up with the actual `depthOrArrayLayers` defaulting to 1. Adding a little strictness to the types here makes sure that you don't accidentally provide `depth`.

Reply:
oh `depth` is `undefined` here. I thought it was `number`

Reply:
looking at this more closely, it only sort of helps. For example this would be invalid:

```ts
    function maketexture(size: GPUExtent3DDict) {
      device.createTexture({ size , format: 'r8unorm', usage: 0, });
    }
    maketexture({ width: 3, height: 3, depth: 3 });
```
but this would still be valid:
```ts
    const sizeconst = { width: 3, height: 3, depth: 3 };
    maketexture(sizeconst);
```